appIndicator.js

```diff
@@ -403,6 +403,13 @@ class AppIndicators_IconActor extends St.Icon {
     }
 
     _createIconByName(path, callback) {
+        if (!path) {
+            GLib.idle_add(GLib.PRIORITY_DEFAULT_IDLE, () => {
```

Comment:
I'm not sure the idle is needed here, can't we just call the callback and return here?

Reply:
Ah, I saw it was failing as per https://github.com/ubuntu/gnome-shell-extension-appindicator/commit/6b5bd5c5d136a9e9f35633ca43c5ce5064af0781 

Not sure there's another way to avoid.

In any case, pelase save the idle id in the class and disconnect it on destruction.

Reply:
I call it from IDLE to ensure that the calling code ends anything that is doing before the callback is called. Currently it doesn't matter, but being strict, the code that calls _createIconByName() can legally presume that the callback won't be called until the whole function ended. Calling the callback directly instead of using idle_add can result in breaking that assumption.

About the id: the callback already returns false to remove it, so it doesn't need to be disconnected on destruction. In fact, it MUST be removed after the first execution, or the callback would be called every time there is idle time.

(unless I understood the memory management wrong in these cases)